### PR TITLE
fix: Checkbox column width should be fixed

### PR DESCRIPTION
**Pull request by harshit078**
## Description

- This PR adds the functionality of fixed column width across all boards
- This PR fixes the issue 
#8463 
#8331

Screenshot 2024-11-14 at 12 19 02 PM

### CODE REVIEW

File: packages/twenty-front/src/modules/object-record/record-table/record-table-header/components/RecordTableHeaderCheckboxColumn.tsx [hunk 1]
```diff
@@ -13,22 +13,30 @@ const StyledContainer = styled.div`
...
+const StyledColumnHeaderCell = styled.th<{ setNumColumns: number }>`
   background-color: ${({ theme }) => theme.background.primary};
   border-bottom: 1px solid ${({ theme }) => theme.border.color.light};
   border-right: transparent;
-  max-width: 30px;
-  min-width: 30px;
-  width: 30px;
+  text-align: center;
+  width: ${({ setNumColumns }) => {
+    return setNumColumns <= 6 ? '22px' : '32px';
```
Comment (FelixMalfait) [inline packages/twenty-front/src/modules/object-record/record-table/record-table-header/components/RecordTableHeaderCheckboxColumn.tsx:22]: Hey @harshit078 thanks! This doesn't seem like something we'd merge (too hacky). No luck with pure-css solutions? That's what we'd want ideally

Comment (harshit078) [reply packages/twenty-front/src/modules/object-record/record-table/record-table-header/components/RecordTableHeaderCheckboxColumn.tsx:22]: Pure-css solution would make the table laggy and not the best approach !

**Comment by harshit078** [comment]
@FelixMalfait, I think this issue is much more deep as currently when the no of columns are less than 4 or 5, the table seems to move away from its position resulting in a un-aligned behaviour. As you can see in notes table, the checkbox column is not aligned. A better way to approach this would be to fix this behaviour.

Screenshot 2024-11-14 at 4 09 34 PM
Screenshot 2024-11-14 at 4 09 51 PM


- Ideal Behaviour
Screenshot 2024-11-14 at 4 11 27 PM

**Comment by FelixMalfait** [comment]
@harshit078 yes this is a CSS issue, not something that should be fixed with JS :)
The problem is with the (+) column, we could remove the 100% width on the table:
Screenshot 2024-11-14 at 12 29 14


Or just set the right column (+)'s width to 100% (width min-width as 32px and changing the alignment / setting a 8px padding around the 16px icon to be 32px)

Screenshot 2024-11-14 at 12 33 25

Let's go with the second option

Thanks!

**Comment by harshit078** [comment]
@FelixMalfait just to be sure, are we setting a width of 100% to the RecordLastColumn instead of 32px ? As that would make the rest of the columns width to be small.


Screenshot 2024-11-14 at 11 58 59 PM

**Comment by FelixMalfait** [comment]
The width of other columns is set to a fixed value (which you can change manually with the mouse), so it shouldn't be a problem, 100% would mean 100% of room left

**Comment by harshit078** [comment]
@FelixMalfait, upon setting the width to be 100% it makes the Dropdown to be hide in Record boards with a lot of fields/columns.

https://github.com/user-attachments/assets/cb707be2-8419-41a3-bc15-912aa90fc1aa

**Comment by FelixMalfait** [comment]
Did you add the minWidth as mentioned?

**Comment by harshit078** [comment]
yes, `width:100%, min-width:32px; `

**Comment by FelixMalfait** [comment]
Could you please push? Thanks

**Review by FelixMalfait — APPROVED** [review]
Thanks! I saw the same behavior you described and ended up using a small hack (adding a second cell)

**Pull request merged**